### PR TITLE
Wrote Two Loss Functions for Training Reward Model

### PR DESCRIPTION
**Pull request by wilderfulford**
- PreferenceCrossEntropyLoss: loss on negative log-likelihood of predicting correct winner

- PreferenceHingeLoss: hinge loss on (winner_reward_score - loser_reward_score)

### CODE REVIEW

File: src/reward_modelling/preference_datasets.py [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+    def __getitem__(self, idx):
+        return self.winner_responses[idx], self.loser_responses[idx]
+
+    def __len__(self):
+        return len(self.winner_responses)
+
+
+ahh = AnthropicHelpfulHarmless()
+print(max(len(ahh[i][1].split()) for i in range(len(ahh))))
```
Comment (mukobi) [inline src/reward_modelling/preference_datasets.py:33]: Can you wrap this in if name == "main"?

Comment (mukobi) [reply src/reward_modelling/preference_datasets.py:33]: Or delete it if no longer necessary

**Review by oliversssf2 — APPROVED** [review]

File: src/reward_modelling/reward_model.py [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+        reward_scores: size (batch, 2) tensor of reward scores,
+            with winner_scores in first column and loser_scores in second.
+
+        """
+        batch_size = inputs['input_ids'].shape[0]
+
+        scores = model(**inputs)
+
+        chosen_scores = scores[:(batch_size//2)]
+        rejected_scores = scores[(batch_size//2):]
```
Comment (mukobi) [inline src/reward_modelling/reward_model.py:83]: A comment would be nice

File: src/reward_modelling/reward_model.py [hunk 3]
```diff
@@ -0,0 +1,182 @@
...
+
+    def __call__(self, batch):
+
+        chosen_responses = []
+        rejected_responses = []
+        responses = []
+        for chosen_response, rejected_response in batch:
+            chosen_responses.append(chosen_response)
+            rejected_responses.append(rejected_response)
+        responses = chosen_responses+rejected_responses
```
Comment (mukobi) [inline src/reward_modelling/reward_model.py:133]: A comment about this batching would be nice

**Review by mukobi — CHANGES_REQUESTED** [review]

**Pull request merged**